src/fusor/models.py

```diff
@@ -77,23 +72,23 @@ class FunctionalDomain(BaseModel):
                 "type": "FunctionalDomain",
                 "status": "lost",
                 "label": "Tyrosine-protein kinase, catalytic domain",
-                "_id": "interpro:IPR020635",
-                "associated_gene": {
+                "id": "interpro:IPR020635",
+                "gene": {
                     "id": "gene:NTRK1",
                     "gene_id": "hgnc:8031",
                     "label": "8031",
-                    "type": "GeneDescriptor",
+                    "type": "Gene",
                 },
-                "sequence_location": {
-                    "id": "fusor.location_descriptor:NP_002520.2",
-                    "type": "LocationDescriptor",
-                    "location": {
-                        "sequence_id": "ga4gh:SQ.vJvm06Wl5J7DXHynR9ksW7IK3_3jlFK6",
-                        "type": "SequenceLocation",
-                        "interval": {
-                            "start": {"type": "Number", "value": 510},
-                            "end": {"type": "Number", "value": 781},
-                        },
+                "sequenceLocation": {
+                    "id": "NP_002520.2",
```

Comment:
if we want `id` to be a `CURIE` then this will need a namespace (`refseq:`)

Reply:
I though we had agreed on using the GA4GH computed identifier for `sequenceLocation.id`. I think the current id should be `sequenceReference.id` and the change the current `sequenceReference.id` to `sequenceReference.label`

Reply:
The current ID is also confusing because in sequenceReference it's listed as a genomic reference sequence, but ID has it as protein reference sequence

Reply:
Hopefully I understood all this correctly - updated in the new PR! Except just realized that this should be a string instead of a CURIE since we want it to be a ga4gh identifier, so I'll fix that. Should I be including the `ga4gh:` prefix on these in case we ever want to support other types for id?

Reply:
wait, nvm, the CURIE id is for this element itself (FunctionalDomain) `"id": "interpro:IPR020635",` so I think it's correct to leave the id for the class param as CURIE. Just need to ensure that the id provided to the passed in `SequenceLocation` in the constructor is correct (ga4gh id) and to ensure that the id passed into `FunctionalDomain` when creating an object is the CURIE